datashape/tests/test_discovery.py

```diff
@@ -292,3 +294,15 @@ def __init__(self, shape, dtype):
 def test_discover_bytes():
     x = b'abcdefg'
     assert discover(x) == String('A')
+
+
+@pytest.mark.xfail(sys.version_info[0] == 2,
+                   raises=NotImplementedError,
+                   reason='xrange cannot be discovered')
+def test_discover_range():
+    assert discover(range_(0)) == Range(int64)
+
+
+def test_discover_slice():
+    assert discover(slice(1)) == Range(int64)
+    assert discover(slice(None)) == Range(float64)  # default
```

Comment:
I guess it is common to use an unbound range (-infinity,infinity) so I wasn't sure what the type of that would be. Maybe just don't discover slices with no types?